lib/Partitioner/PartitionerUtils.cpp

```diff
@@ -156,6 +156,20 @@ uint64_t getOutMemPerNode(const NodesSetTy &nodes, const Node *node) {
   return ret;
 }
 
+/// Given a node, \return the NodeSet of all nodes that create the results
+/// for any of the inputs of this node (i.e. input of inputs)
+NodesSetTy getInputs(const Node *node) {
+  NodesSetTy result;
+  for (size_t i = 0, e = node->getNumInputs(); i < e; i++) {
+    Node *input = node->getNthInput(i).getNode();
+    Storage *in = llvm::dyn_cast<Storage>(input);
+    if (!in) {
```

Comment:
Could shorten this a bit as `if (!llvm::isa<Storage>(input))` but no big deal either way.